misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -196,6 +271,22 @@
                         generatedFunctionName =
                                 String.format("map%sTo%s(%s)", foundTypeRight, foundTypeLeft, functionCall);
                         fEdits.add(new TextEdit(newTextRange, generatedFunctionName));
+                    } else if (foundErrorLeft && !foundErrorRight) {
+                        String checkExpression = "";
+                        Collection<ChildNodeEntry> childEntries = matchedNode.childEntries();
+                        for (ChildNodeEntry childEntry : childEntries) {
+                            if (childEntry.name().equals("expression")) {
+                                functionCall = childEntry.node().isPresent() ?
+                                        childEntry.node().get().toString() : null;
+                            } else if (childEntry.name().equals("expression")) {
+                                checkExpression = childEntry.node().isPresent() ?
+                                        childEntry.node().get().toString() : null;
+                            }
```

Comment:
Why do we use the same expression here? Only the if clause is executed. Revamp this logic

Reply:
changed the logic